Update Theme articles with Application Theme (#41)
vaadin/docs: articles/themes/importing-style-sheets.asciidoc

[.example]
--
In server-side views (Java), the standard way is to organize style sheets as an <<application-theme, application theme>> then use `@Theme` annotation to apply the application theme.

Otherwise, you can also use the https://vaadin.com/api/platform/com/vaadin/flow/component/dependency/CssImport.html[`@CssImport`] annotation to import local/bundled style sheets and the `@StyleSheet` annotation to import external/linked style sheets.
The `@CssImport` annotation can also be used to import component-specific style sheets.

[source, java]
----
include::src/main/java/com/vaadin/flow/tutorial/theme/ImportingStyleSheets.java[tags=cssimport;cssimport-themefor;stylesheet;app-class, indent=0, group=Java]
----

> NOTE: Using an <<application-theme, application theme>>, a component-specific style sheet is imported when the CSS file name matches the component name. For example,  `my-theme/components/vaadin-button.css` will be the same as `@CssImport` with `themeFor="vaadin-button"`.

Reviewer: > `my-theme/components/vaadin-button.css`
Perhaps it should have the full path `frontent/theme/my-theme/components/vaadin-button.css`

Author: Fixed.

> NOTE: Using an <<application-theme, application theme>>, a component-specific style sheet is imported when the CSS file name matches the component name. For example, `frontent/theme/my-theme/components/vaadin-button.css` will be the same as `@CssImport` with `themeFor="vaadin-button"`.

Reviewer 2: Otherwise, it the CSS file name does not match the component tag, the CSS would not be recognised and not be applied?

> NOTE: Using an <<application-theme, application theme>>, component-specific style sheets are imported and will be applied to components that have their tag names matches with the CSS file names. For example, `frontent/theme/my-theme/components/vaadin-button.css` will be the same as `@CssImport` with `themeFor="vaadin-button"`.

Reviewer: ```suggestion
NOTE: When using an <<application-theme, application theme>>, component-specific style sheets are imported from the components folder and will be applied to components where the CSS file name match the component tag. For example, `frontent/theme/my-theme/components/vaadin-button.css` will be the same as `@CssImport` with `themeFor="vaadin-button"`.
```
